app/controllers/admin/decisions_controller.rb

```diff
@@ -1,19 +1,19 @@
 class Admin::DecisionsController < AuthenticatedAdminController
   def new
     @split = Split.find params[:split_id]
-    @decision = Decision.new
+    @decision = @split.build_decision
   end
 
   def create
     split = Split.find params[:split_id]
-    CreateDecisionJob.perform_later(split, variant: target_variant, admin: current_admin)
-    flash[:success] = "Queued decision to reassign all visitors to #{target_variant}"
+    split.create_decision!(decision_params)
```

Comment:
Something about having surface area on a model for a _different_ model's persistence operations feels strange (even though I know that Rails autogenerates the same `create_` methods for AR associations). If the method didn't already exist, I'd suggest using `@split.build_decision` directly (and calling `#save!` on the result).

Reply:
yeah, I'm not married to it, it feels a little railsy on the outside, but it's not doing anything for keeping split from becoming a god model.

Reply:
One nice thing in retrospect is that `create_decision!` reloads split based on the decision, so it kind of is related.